ofrak_core/ofrak/gui/server.py

```diff
@@ -651,6 +653,22 @@ async def get_all_tags(self, request: Request) -> Response:
             self._serializer.to_pjson(self._ofrak_context.get_all_tags(), Set[ResourceTag])
         )
 
+    @exceptions_to_http(SerializedError)
+    async def flush_to_disk(self, request: Request) -> Response:
+        resource = await self._get_resource_for_request(request)
+        output_file_name = self._serializer.from_pjson(await request.json(), str)
+        # Use FilesystemEntry name as filename if available, otherwise generate random filename
+        if not output_file_name:
+            output_file_name = str(uuid.uuid4())
```

Comment:
We have an "ID service" that might be useful for doing this. It usually generates UUIDs under the hood, but can also generate sequential IDs – it's what's used for generating the resource IDs. It would be worth spending a quick 15 minutes to see if it would be easy to integrate here. If not, no need to let that block the PR.